addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java

```diff
@@ -748,15 +750,15 @@ private ThingStatusDetail authenticate(String username, String password) {
 
         Response response = tokenTarget.request().post(Entity.entity(payLoad, MediaType.APPLICATION_JSON_TYPE));
 
-        logger.debug("Authenticating : Response : {}:{}", response.getStatus(), response.getStatusInfo());
-
         if (response != null) {
+            logger.debug("Authenticating : Response : {}:{}", response.getStatus(), response.getStatusInfo());
```

Comment:
The old case would throw an NPE in the debug log statement before reaching the if.